packages/framework/aqueduct/src/data-object-factories/dataObjectFactory.ts

```diff
@@ -24,16 +24,16 @@ import { PureDataObjectFactory } from "./pureDataObjectFactory";
  * ensuring relevant shared objects etc are available to the factory.
  *
  * Generics:
- * P - represents a type that will define optional providers that will be injected
+ * O - represents a type that will define optional providers that will be injected
  * S - the initial state type that the produced data object may take during creation
  */
-export class DataObjectFactory<TObj extends DataObject<P, S>, P, S> extends PureDataObjectFactory<TObj, P, S>
```

Comment:
fwiw it looks like the `O` usage elsewhere is to distinguish between optional and required providers, whereas here there's no distinction which is why this was `P` instead.  imo `P` would still make more sense here but not a big deal either way

Reply:
Can you please elaborate? The comment above says P - is for optional providers.
Digging deeper, it maps to 
```
protected readonly providers: AsyncFluidObjectProvider<FluidObjectKey<P>, FluidObjectKey<object>>;

export type AsyncFluidObjectProvider<O extends keyof IFluidObject, R extends keyof IFluidObject>
    = AsyncOptionalFluidObjectProvider<O> & AsyncRequiredFluidObjectProvider<R>;
```
as well as 
```
    const providers = dependencyContainer.synthesize<P>(this.optionalProviders, {});

    public synthesize<
        O extends IFluidObject,
        R extends IFluidObject = {}>(
            optionalTypes: FluidObjectSymbolProvider<O>,
            requiredTypes: FluidObjectSymbolProvider<R>,
    ): AsyncFluidObjectProvider<FluidObjectKey<O>, FluidObjectKey<R>> {

```
So it's consistent through code base P stands for optional providers.
Did I miss something?

Reply:
didn't miss anything, i uh... just didn't really have anything substantial to say i guess 😐